### DISCUSSION TITLE
Status not reliable

### DISCUSSION THREAD

**Discussion by DJFlaxi** [philibertc/micronova_controller#33] [Q&A]
Hello,

I have everything up and running now.
However, I still have two minor problems.

1. the status is not updated
For example, when I turn off, the status shows "turn off" but then does not update to "Off".

2. to switch on or off I have to send the command "ON" or "OFF" twice in succession.

Can anyone help me with this?

Sorry for my bad english. That's why Deepl helped me :-)

**Comment by philibertc** [comment]
Hello,

1. Does the stove's display show off or still turning off as well?
2. Hmm, that's strange. Can you send the logs when you send these commands, readable from the serial port of the esp (115200 baud).

Best regards,
philibertc

**Comment by DJFlaxi** [comment]
Hi,

1. in the log it don not display the Status of the stove:
`12:20:01.674 -> Attempting MQTT connection...connected
12:20:23.223 -> Param=1 value=24 T. amb. 18.00
12:20:23.429 -> Param=3e value=15 T. fumes 21
12:20:23.637 -> Param=34 value=0 Fire 0
12:20:48.460 -> Param=0 value=6 Param=34 value=0 Fire 0
12:21:38.243 -> Param=1 value=24 T. amb. 18.00
12:21:38.451 -> Param=3e value=16 T. fumes 22
12:21:38.652 -> Param=34 value=0 Fire 0
12:22:03.459 -> Param=3e value=15 T. fumes 21
12:22:03.662 -> Param=34 value=0 Fire 0`

I have changed the Code and set the Status of the stove to the second Position:
`void getStates() //Calls all the get…() functions
{
    getAmbTemp();
    delay(500);
    getStoveState();
    delay(100);
    getAmbTemp();
    delay(100);
    getFumeTemp();
    delay(100);
    getFlamePower();
    if (int_hydro_mode == 1)
    {
        delay(100);
        getWaterTemp();
        delay(100);
        /*getWaterSet();
        delay(100);*/
        getWaterPres();
    }
}`
then it works.
But I have to get states the `getAmbTemp();` twice.

2. This is the log when I send the "ON" command:
`12:24:40.378 -> Attempting MQTT connection...connected
12:24:58.656 -> Param=34 value=0 Fire 0
12:25:48.271 -> Param=5 value=0 Param=3e value=15 T. fumes 21
12:25:48.677 -> Param=34 value=0 Fire 0
12:25:55.623 -> Message arrived [micronova/intopic] ON
12:25:57.344 -> Param=34 value=0 Fire 0
12:26:13.496 -> Param=f9 value=6 Param=34 value=0 Fire 0`

But nothing happens.

**Reply by DJFlaxi** [reply]
I would have possibly an idea, how one can guarantee the switching on and/or switching off.
One would have to query the status before the switching command comes. Unfortunately, I do not know how to program this.

Can you program this for me on a test basis?

Thanks

**Reply by philibertc** [reply]
Hello,

Yes, I can do that. There are already a few tests here, including one that turns on and then 2 seconds later turns off the stove, but without reading the status: https://github.com/philibertc/micronova_controller/tree/master/tests/esp8266

Best Regards,
Philibert

**Reply by philibertc** [reply]
I'll think about what could be going on, thanks to the logs above.

**Comment by DJFlaxi** [comment]
Hello,

i tested the test1_on_off.ino.

The Stove is "on".
When I connected is with the test1_on_off.ino, the Stove is turning off but not on again.

**Reply by philibertc** [reply]
That's normal: the test turns on then off, not off then on.

**Comment by DJFlaxi** [comment]
Hi,

i have tested the test1_on_off.ino.
It works. The Stove turn on and two seconds later it turns off.

The commands are right.
I think the Problem is the MQTT-Command to set the Stove on or off.

**Reply by DJFlaxi** [reply]
Hi @philibertc,
do you have a soloution for me?

**Reply by philibertc** [reply]
Hello,
I'm working on it. The problem is clearly with the MQTT part, and I'm trying to find out from the shared logs exactly what function is causing the problem.

Best Regards,
Philibert

**Reply by DJFlaxi** [reply]
Thank you for your work
